packages/Search/test/tstDetailsAlgorithms.cpp

```diff
@@ -152,3 +152,29 @@ TEUCHOS_UNIT_TEST( DetailsAlgorithms, centroid )
     TEST_EQUALITY( centroid[1], 5.0 );
     TEST_EQUALITY( centroid[2], 15.0 );
 }
+
+TEUCHOS_UNIT_TEST( DetailsAlgorithms, is_valid )
+{
+    using DataTransferKit::Box;
+    using DataTransferKit::Details::isValid;
+    using DataTransferKit::Point;
+    using DataTransferKit::Sphere;
+
+    auto const infty = std::numeric_limits<double>::infinity();
+
+    TEST_ASSERT( isValid( Point{{1., 2., 3.}} ) );
+    TEST_ASSERT( !isValid( Point{{0., infty, 0.}} ) );
+
+    TEST_ASSERT( isValid( Box{{{1., 2., 3.}}, {{4., 5., 6.}}} ) );
+    TEST_ASSERT( isValid( Box{{{0., 0., 0.}}, {{0., 0., 0.}}} ) );
+    TEST_ASSERT( !isValid( Box{{{0., 0., -infty}}, {{0., 0., 0.}}} ) );
+    TEST_ASSERT( !isValid( Box{{{0., 0., 0.}}, {{+infty, 0., 0.}}} ) );
+    TEST_ASSERT( isValid( Box{} ) );
+
+    TEST_ASSERT( isValid( Sphere{{{1., 2., 3.}}, 4.} ) );
+    TEST_ASSERT( !isValid( Sphere{{{0., 0., 0.}}, 0.} ) );
```

Comment:
Why is this valid for a `Box` but not for a `Sphere`?

Reply:
radius has to greater than zero

Reply:
Why?

Reply:
We can change to greater than or to zero, that's fine with me

Reply:
I think it makes sense to have the same behavior for `Box` and `Sphere`. If I can build a `Box` of size zero, I should be able to build a `Sphere` of radius zero (unless we have a good reason).

Reply:
Agreed.  Will change.